contracts/metatx/MinimalForwarder.sol

```diff
@@ -88,17 +119,50 @@ contract MinimalForwarder is EIP712 {
         return (success, returndata);
     }
 
-    function _recover(ForwardRequest calldata req, bytes calldata signature) internal view returns (address) {
+    /**
+     * @dev Validates if the provided request can be executed at `blockNumber` with `signature`.
+     */
+    function _validateAt(
```

Comment:
What was the reasoning to separate `_validate` and `_validateAt`? I would keep just one, to simplify.

Reply:
None in particular